src/linear_algebra/basic.lean

```diff
@@ -104,6 +104,22 @@ linear_map.ext $ λ x, rfl
 theorem comp_assoc (g : M₂ →ₗ[R] M₃) (h : M₃ →ₗ[R] M₄) : (h.comp g).comp f = h.comp (g.comp f) :=
 rfl
 
+section
+variables (R M)
+
+/--
+The type-level equivalence between R-linear maps from `R` to `M`, and points of `M` itself.
+This says that the forgetful functor from `R`-modules to types is representable, by `R`.
+-/
+@[simps]
+def ring_lmap_equiv_self : (R →ₗ[R] M) ≃ M :=
+{ to_fun := λ f, f 1,
+  inv_fun := λ m, { map_smul' := λ r r', by simp [mul_smul], ..(smul_add_hom R M).flip m },
+  left_inv := λ x, by { ext, simp },
+  right_inv := λ x, by { simp } }
```

Comment:
This is probably a `≃ₗ[S]`, right?

Reply:
Yes. Is it worth mentioning an otherwise uninvolved `S`? Why not just lift it to an `R`-linear map?

Reply:
Ah, I've read the library note now. :-) Nifty! I will update this.

Reply:
because when `R` is not commutative, you can't even state this as a `≃ₗ[R]`.

Reply:
Ah, okay, I've read the library note: nifty! I will update this.